File: common_base_ui/src/main/java/com/fido/common/common_base_ui/base/listener/BlockQuickClickListener.java
package com.fido.common.common_base_ui.base.listener;

import android.view.View;

public abstract class BlockQuickClickListener implements View.OnClickListener {
    private static final long MULTI_CLICK_INTERVAL = 800L;
    private long lastClickTime;

    public abstract void onBlockQuickClick(View v);

    @Override
    public void onClick(View v) {
        if (isFastClick()) return;
        onBlockQuickClick(v);
    }

    public boolean isFastClick() {
        long currentTime = System.currentTimeMillis();
        if (currentTime - lastClickTime < MULTI_CLICK_INTERVAL) {
            return true;
        } else {
            lastClickTime = currentTime;
            return false;
        }
    }
}

File: common_base_util/src/main/java/com/fido/common/common_base_util/listener/BlockQuickClickListener.kt
package com.fido.common.common_base_util.listener

import android.view.View

abstract class BlockQuickClickListener(private val interval:Long = 800L):View.OnClickListener {

    private var lastClickTime: Long = 0

    abstract fun onBlockQuickClick(v: View)

    override fun onClick(v: View) {
        if (isFastClick()) return
        onBlockQuickClick(v)
    }

    private fun isFastClick(): Boolean {
        val currentTime = System.currentTimeMillis()
        return if (currentTime - lastClickTime < interval) {
            true
        } else {
            lastClickTime = currentTime
            false
        }
    }

}